### DISCUSSION TITLE
cascade ISourceGenerator

### DISCUSSION THREAD

**Discussion by hopperpl** [dotnet/roslyn#67213] [Q&A] [answered]
I have a C# source generator (`ISourceGenerator`) that performs native API binding code generation from **yaml** files.

I would like to facilitate `LibraryImportGenerator` on the generated C# source text. Is there a way to do so?

I also just noticed that `ISourceGenerator` has been deprecated in favor of `IIncrementalGenerator` and that looks like what I will need. And finally comes with caching support.

Would this approach work for me and are there any samples how to do so?

*  `one.yaml` -> `one.g.cs` -> ImportLibraryGenerator applied on provided source `one.g.cs`
*  `two.yaml` -> `two.g.cs` -> ImportLibraryGenerator applied on provided source `two.g.cs`
*  ...

**Comment by CyrusNajmabadi** [comment, marked as answer]
> Would this approach work for me and are there any samples how to do so?

No. Source generators cannot see the results of another generator. Sorry!

**Comment by sharwell** [comment]
Duplicate of #57239

**Reply by b-straub** [reply]
@CyrusNajmabadi 
Is their any attempt to find a solution for this problem. The runtime is introducing new generators REGEX, JSON (AOT, trimming, performance reasons) and those can not be used inside an another generator.
Another very desirable use case might be the generation of RAZOR files.
https://github.com/dotnet/roslyn/issues/57239#issuecomment-1545651854

**Reply by sharwell** [reply]
@b-straub This issue is closed as a duplicate of #57239. Please follow up there if you have any further questions. Thanks! 😄
